tests/foreman/virtwho/test_ui_virtwho.py

```diff
@@ -420,3 +427,155 @@ def test_positive_virtwho_configs_widget(session, form_data):
         assert not session.virtwho_configure.search(name)
         session.organization.select("Default Organization")
         session.organization.delete(org_name)
+
+
+@tier2
+def test_positive_delete_config_delete_user(session, form_data):
+    """Verify when a config is deleted the associated user is deleted.
+
+    :id: 9fdee7f2-833c-47e0-9d58-cd0c9fdd15fe
+
+    :steps:
+        1. Create a virt-who configuration and deploy it to a
+           virt-who server.
+        2. Delete the configuration on the Satellite.
+
+    :expectedresults:
+        1. Verify the virt-who server can no longer connect to the
+           Satellite.
+
+    """
+    name = gen_string('alpha')
+    form_data['name'] = name
+    with session:
+        session.virtwho_configure.create(form_data)
+        config_id = get_configure_id(name)
+        config_command = get_configure_command(config_id)
+        deploy_configure_by_command(config_command)
+        assert session.virtwho_configure.search(name)[0]['Status'] == 'ok'
+        session.virtwho_configure.delete(name)
+        assert not session.virtwho_configure.search(name)
+        restart_virt_who_service()
+        try:
+            deploy_validation()
+        except Exception as Error:
+            assert str(Error) == 'Failed to start virt-who service'
+
+
+@tier2
+def test_positive_virt_who_roles_permissions(session, test_name, form_data):
+    """Verify the virt-who roles can TRULY work.
+
+    :id: 4bef3b31-89d2-4502-b971-2437b920a3ff
+
+    :expectedresults:
+        Permissions:
+        1. Virt-who Reporter Role granting minimal set of permissions for virt-who
+        to upload the report, it can be used if you configure virt-who manually
+        and want to use user that has locked down account.
+        2. Virt-who Viewer Role granting permissions to see all configurations
+        including their configuration scripts, which means viewers could still
+        deploy the virt-who instances for existing virt-who configurations.
+        3. Virt-who Manager Role granting all permissions to manage virt-who configurations,
+        user needs this role to create, delete or update configurations.
+
+    """
+    username = gen_string('alpha')
+    password = gen_string('alpha')
+    with session:
+        # Create a virt-who config plugin
+        virt_who_name = gen_string('alpha')
+        form_data['name'] = virt_who_name
+        session.virtwho_configure.create(form_data)
+        session.virtwho_configure.read(virt_who_name)
+        # Create an user
+        session.user.create({
+            'user.login': username,
+            'user.mail': 'test@test.com',
+            'user.auth': 'INTERNAL',
+            'user.password': password,
+            'user.confirm': password,
+        })
+        # Create a virt-who config file manually
+        virtwho_cleanup()
+        create_etc_d_file(form_data, username, password)
```

Comment:
As discussion, don't need to create the configure file by manual,  it will be created by `session.virtwho_configure.create(form_data)`, you just need to update it.

Reply:
yes, will modify this with next commit